modin/experimental/xgboost/xgboost.py

```diff
@@ -25,7 +25,7 @@
 LOGGER = logging.getLogger("[modin.xgboost]")
 
 
-class DMatrix(xgb.DMatrix):
```

Comment:
Please make this change in a separate pull request, as it does not apply to the current one.

Reply:
If I leave only changes related to fix issue, the new issue will occur - `doc_checker` will found errors for `label` parameter, because in that case docstring from native `xgb.DMatrix` (has differences from our style) will be used. Does it make sense to move inheritance to a separate RR?

Reply:
In this case, `doc_checker` should have found errors before this pull request. How did it work before?

Reply:
In my assumption, inheritance of docstrings happens only in case `label=None` (when we aligned with parent class). Before it was positional parameter with own docstring.

Reply:
I think we should fix `doc_checker` in this case as users might rely on inheritance.

Reply:
The root cause of issue is in incorrect docstring of `__init__` method in parent class (`xgb.DMatrix`) for `label` parameter. In our code we skip the check of correctness of inherited docstrings with decorator `_inherit_docstrings`. I don't see the point to add this decorator to the method `__init__`, because we don't require separate docstrings for `__init__` methods in our docstyle. So, I just added `noqa` to skip the check of doc here.

Reply:
If the documentation does inherit for `__init__`, then it's a problem because it [doesn't match](https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.DMatrix) the function signature. We need to delete it.

Reply:
I added decorator `_remove_docstring`, that removes docstring for object (for `__init__` method in our case).